### PR TITLE
Upgrade to net6.0 and update dependencies

### PR DESCRIPTION
**Pull request by xoofx**
Will probably conflict with #2 but gonna make a separate PR for it

### CODE REVIEW

File: global.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
+    "sdk": {
```
Comment (daveaglick) [inline global.json:2]: I generally don't pin my SDK versions, but I'm not opposed to it either if there's a reason to. Are we using something in this project that benefits from pinning and might change with later SDKs, or just adding this as a best practice?

**Comment by daveaglick** [comment]
This is definitely preferable to #2, at least as far as the target updates go. Updating to .NET Standard 2 seems reasonable (#2 goes up to .NET 5), and I'd rather keep these targets as low as possible for this one since it's a utility library. That one also needs some other work, so I'm going to merge this PR and then keep iterating on #2.

File: global.json (hunk 1 above)
Comment (xoofx) [reply global.json:2]: So, it's to ensure more predictability with local builds. On CI you can control which version is exactly used, not on local builds. But from past experience, I have found some users and also myself had installed a recent .NET preview that could break entirely the build of a project. So usually, I don't like to chase down this kind of issues 😅

Comment (daveaglick) [reply global.json:2]: Figured that was it, but wanted to be sure I wasn't missing some crazy version compatibility thing that requires a specific SDK to build. Glad it's just plain old .NET build tooling touchiness!

**Pull request merged**